java/clients/src/main/java/sleeper/status/report/filestatus/StateStoreFiles.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.status.report.filestatus;
+
+import sleeper.core.partition.Partition;
+import sleeper.statestore.FileInfo;
+import sleeper.statestore.StateStore;
+import sleeper.statestore.StateStoreException;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public class StateStoreFiles {
```

Comment:
Not sure this is the right name, as it contains the partitions as well. It's arguably a cache of the contents of the StateStore?

Reply:
I've renamed it to StateStoreSnapshot.